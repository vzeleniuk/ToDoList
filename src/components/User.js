import React from 'react';
import PropTypes from 'prop-types';

export class User extends React.Component {
  render() {
    return(
      <div className="container">
        <div className="row">
          <p>User Name</p>
          <button onClick={this.props.greet} className="btn btn-primary">Greet</button>
        </div>
      </div>
    )
  }
}

User.propTypes = {
  greet: PropTypes.func
}