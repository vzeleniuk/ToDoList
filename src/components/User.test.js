import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import User from './User';

describe('<User />', () => {
  test('User component should render as expected', () => {
    const wrapper = shallow(<User />);
    const tree = toJson(wrapper)
    expect(tree).toMatchSnapshot();
  })

  test('Simulate click event', () => {
    window.alert = jest.fn();
    const wrapper = mount(<User/>);
    const button = wrapper.find('button');
    button.simulate('click');
    expect(window.alert).toHaveBeenCalledWith('Hello, User!');

  })
})


// describe('User component should render as expected', () => {
//   describe('<User />', () => {
//     it('renders the component', () => {
//       const component = shallow(<User />);
//       // const component = component.dive();
//       console.log(component);
//       // expect(component.find('h1').hasClass('cover-heading')).toBe(true)
//       // expect(toJson(component)).toMatchSnapshot();
//     })

//     // it('successfully calls the onClick handler', () => {
//     //   const mockOnClick = jest.fn();
//     //   const component = shallow(
//     //     <GatorButton onClick={mockOnClick} label="Eat Food" />
//     //   );
//     //   const component = component.dive();

//     //   component.find('button').simulate('click');

//     //   expect(mockOnClick.mock.calls.length).toEqual(1);
//     })
//   });
// });