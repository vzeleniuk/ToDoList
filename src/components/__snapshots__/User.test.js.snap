// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<User /> User component should render as expected 1`] = `
<div
  className="container-main"
>
  <div
    className="row"
  >
    <h1
      className="cover-heading mt-4 mb-4"
    >
      User Name
    </h1>
  </div>
  <div
    className="row"
  >
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Greet
    </button>
  </div>
</div>
`;
